### PR TITLE
ci(jenkins): enable release process

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- Enable ITs: async when running as a PR or sync when running in a branch/tag
- Continuous delivery when running the pipeline either in a branch or tag. master branch refers to `latest` while others will name the image with their names.
- Script to bump versions for the python agent.

## Tasks
- [x] Versioning
- [x] Release automation
- [x] ITs validation should be synchronous when running a release.

## Issues
- Depends on:
  - https://github.com/elastic/apm-integration-testing/pull/556
  - https://github.com/elastic/apm-pipeline-library/pull/271
  - https://github.com/elastic/apm-agent-python/pull/656

### CODE REVIEW

File: .ci/bump-version.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+AGENT_VERSION="${1?Missing the APM python agent version}"
+
+sed -ibck "s#elastic-apm==.*#elastic-apm==${AGENT_VERSION}#g" requirements.txt
```
Comment (v1v) [inline .ci/bump-version.sh]: pip-compile isn't a standard tool, this is the sed approach for the time being.

File: .ci/bump-version.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+# Commit changes
+git add requirements.txt
+git commit -m "Bump version ${AGENT_VERSION}"
```
Comment (beniwohli) [inline .ci/bump-version.sh]: doesn't this have to be pushed to elastic/opbeans-python? Or does that happen elsewhere?

Comment (beniwohli) [reply .ci/bump-version.sh]: Ah, I guess "elsewhere" is here

https://github.com/elastic/apm-agent-python/pull/656/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R272

Comment (v1v) [reply .ci/bump-version.sh]: Good point, I decoupled the commit from the push. The main reason is about the responsibility, the one in charge for the bump should be caring about committing the new files. Therefore the pipeline orchestrator in the `apm-agent-python` should care for calling this script and push, see https://github.com/elastic/apm-agent-python/pull/656/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R270-R274

**Pull request merged**